app/src/org/commcare/activities/connect/HQUserInviteActivity.java

```diff
@@ -0,0 +1,173 @@
+        finish();
+    }
+
+    private void handleInvitation(String callBackUrl, String inviteCode) {
+        Logger.log("HQInvite", "User accepted invitation");
+
+        IApiCallback callback = new IApiCallback() {
+            @Override
+            public void processSuccess(int responseCode, InputStream responseData) {
+                FirebaseAnalyticsUtil.reportHQInvitationResponse(domain, true, "");
+                Logger.log("HQInvite", "Acceptance succeeded");
+                binding.progressBar.setVisibility(View.GONE);
+                try {
+                    String responseAsString = new String(StreamsUtil.inputStreamToByteArray(responseData));
+                    if (responseAsString.length() > 0) {
+                        startActivity(new Intent(HQUserInviteActivity.this, DispatchActivity.class));
+                        finish();
+                    }
+                } catch (IOException e) {
+                    CrashUtil.reportException(e);
+                    setErrorMessage(getString(R.string.connect_hq_invitation_accept_error));
+                }
+            }
+
+            @Override
+            public void processFailure(int responseCode, IOException e) {
+                FirebaseAnalyticsUtil.reportHQInvitationResponse(domain, false, "API error");
+                Logger.log("HQInvite", "Acceptance failed");
+                binding.progressBar.setVisibility(View.GONE);
+                setErrorMessage(getString(R.string.connect_hq_invitation_accept_error));
+            }
+
+            @Override
+            public void processNetworkFailure() {
+                binding.progressBar.setVisibility(View.GONE);
+                setErrorMessage(getString(R.string.recovery_network_unavailable));
+            }
+
+            @Override
+            public void processOldApiError() {
+                binding.progressBar.setVisibility(View.GONE);
+                setErrorMessage(getString(R.string.recovery_network_outdated));
+            }
+        };
+        ConnectUserRecord user = ConnectManager.getUser(this);
+        binding.progressBar.setVisibility(View.VISIBLE);
+        boolean isBusy = !ApiConnectId.hqUserInvitation(HQUserInviteActivity.this,user.getUserId(),user.getPassword(), callBackUrl, inviteCode, callback);
+        if (isBusy) {
+            Toast.makeText(HQUserInviteActivity.this, R.string.busy_message, Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    public void setErrorMessage(String message) {
+        boolean show = message != null;
+
+        binding.connectPhoneVerifyError.setVisibility(show ? View.VISIBLE : View.GONE);
+
+        if (show) {
+            binding.connectPhoneVerifyError.setText(message);
+        }
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        binding = null;
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
```

Comment:
I don't see us calling `startActivityForResult` anywhere in this class, as such I am confused when and how will this get called ?